tools/importer-rest-api-specs/components/dataapigeneratorjson/templater_terraform_resource_schema.go

```diff
@@ -111,14 +88,14 @@ func topLevelFieldObjectDefinition(input resourcemanager.TerraformSchemaFieldObj
 	return input
 }
 
-func objectDefinitionfromSchemaField(input resourcemanager.TerraformSchemaFieldObjectDefinition) *SchemaFieldObjectDefinition {
-	objectDefinition := SchemaFieldObjectDefinition{
+func objectDefinitionfromSchemaField(input resourcemanager.TerraformSchemaFieldObjectDefinition) *dataApiModels.ObjectDefinition {
+	objectDefinition := dataApiModels.ObjectDefinition{
 		ReferenceName: input.ReferenceName,
-		Type:          TerraformSchemaFieldType(input.Type),
+		Type:          dataApiModels.ObjectDefinitionType(input.Type),
```

Comment:
this assumes these are 1:1 compatible, when the backing values could be different - can we instead define a map for these and pull out the value (`dataApiModels.ObjectDefinitionType`) from the key (`resourcemanager.TerraformSchemaFieldObjectDefinitionType`)